### PR TITLE
Add support for single category layer legend items (ref: missing category and title)

### LINKED ISSUE #377: Missing layer single category/category title
### Checklist

- [X] I've searched through the existing [issues](https://github.com/g3w-suite/g3w-client/issues) and this bug has never been reported before

### Subject of the issue

In case of layer with a single Category

1. Legend On TOC

  Screenshot from 2023-03-14 11-16-27
  No Category Title is show

2. Legend on Separate Tab

  Screenshot from 2023-03-14 11-13-29
  
  No Category is how

### Steps to reproduce

1. Create a Single Category on layer in QGIS project

### Environment

- g3w-admin: 3.5
- g3w-client: 3.7.x
- browser: all

### Link to your project

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by volterra79**
Closes: #377

### CODE REVIEW

File: src/app/core/layers/geolayermixin.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ proto.setCategories = function(categories=[]) {
   this.legendCategories[this.getCurrentStyle().name] = categories;
   //set categories state attribute to true only if exist at least a rule key
   // meaning that layer has at least more than one has a
-  this.state.categories = categories.length > 1 && categories.filter(category => category.ruleKey).length > 1;
+  this.state.categories = categories.length > 0 && categories.filter(category => category.ruleKey).length > 0;
```
Comment (Raruto) [inline src/app/core/layers/geolayermixin.js]: double check the comment above

**Comment by Raruto** [comment]
@volterra79 for me we can merge it (I haven't tested, but it seems quite reasonable).

BTW, there was a particular reason why before the `length` check was set to greater than one?

**Comment by volterra79** [comment]
> @volterra79 for me we can merge it (I haven't tested, but it seems quite reasonable).
> 
> BTW, there was a particular reason why before the `length` check was set to greater than one?

@Raruto it was related to (i don't remember) old way to send categories from a server

File: src/app/core/layers/geolayermixin.js (hunk 1 above)
Comment (volterra79) [reply src/app/core/layers/geolayermixin.js]: @Raruto done

**Comment by Raruto** [comment]
> it was related to (i don't remember) 

@volterra79 it was added in here: https://github.com/g3w-suite/g3w-client/pull/258

**Comment by volterra79** [comment]
> > it was related to (i don't remember)
> 
> @volterra79 it was added in here: #258

@Raruto it was an issue on g3w-admin code

**Comment by Raruto** [comment]
> it was related to (i don't remember)
> > @volterra79 it was added in here: https://github.com/g3w-suite/g3w-client/pull/258
> > > @raruto it was an issue on g3w-admin code

@volterra79 ok, so before merge this one, I would let @leolami test https://github.com/g3w-suite/g3w-client/pull/378 in order to confirm that it will not break: https://github.com/g3w-suite/g3w-client/pull/258

**Comment by volterra79** [comment]
> > it was related to (i don't remember)
> > > @volterra79 it was added in here: #258
> > > > @Raruto it was an issue on g3w-admin code
> 
> @volterra79 ok, so before merge this one, I would let @leolami test #378 in order to confirm that it will not break: #258

@Raruto i checked and it works

File: src/components/CatalogLayerLegend.vue [hunk 2]
```diff
@@ -151,56 +147,68 @@
...
+            if (all) { // case of all categories
+              this._setAllLayerCategories(nodes);
+            } else {
+              this._setLayerCategories(nodes, categories);
             }
           }
-
         } catch(err) {
           this.setError();
         }
```
Comment (Raruto) [inline src/components/CatalogLayerLegend.vue:159]: be aware, here I've removed a nested `try..catch` (I hope it was superfluous)

File: src/components/CatalogLayerLegend.vue [hunk 3]
```diff
@@ -151,56 +147,68 @@
...
-
-              }
-
-            } catch(err) {
-              this.setError();
+            const { nodes = [] } = await projectLayer.getLegendGraphic({ all });
+            if (all) { // case of all categories
+              this._setAllLayerCategories(nodes);
+            } else {
+              this._setLayerCategories(nodes, categories);
```
Comment (Raruto) [inline src/components/CatalogLayerLegend.vue]: Can you find a more meaningful name for the two functions: `_setAllLayerCategories()` and `_setAllLayerCategories()` (for me it's not so not clear what the difference between the two `if-else` conditions)

Comment (volterra79) [reply src/components/CatalogLayerLegend.vue]: @Raruto done

**Review by Raruto — APPROVED** [review]

**Pull request merged**